OpenSim/Simulation/SimbodyEngine/Joint.cpp

```diff
@@ -56,29 +57,69 @@ Joint::Joint() : Super()
-    const PhysicalFrame& child, 
-    const SimTK::Vec3& locationInChild, const SimTK::Vec3& orientationInChild, 
-    bool reverse) : Super()
+/* API constructor. */
+Joint::Joint(const std::string &name, const std::string& parentName, 
+                                      const std::string& childName,
+                                      bool reverse) : Super()
 {
     setNull();
     constructInfrastructure();
-
-    set_location_in_parent(locationInParent);
-    set_orientation_in_parent(orientationInParent);
-    set_location_in_child(locationInChild);
-    set_orientation_in_child(orientationInChild);
     set_reverse(reverse);
 
-    updConnector<PhysicalFrame>("parent_body").set_connectee_name(parent.getName());
-    updConnector<PhysicalFrame>("child_body").set_connectee_name(child.getName());
+    updConnector<PhysicalFrame>("parent_frame").set_connectee_name(parentName);
+    updConnector<PhysicalFrame>("child_frame").set_connectee_name(childName);
+
+    setName(name);
+}
 
+/* Deprecated Constructor*/
+Joint::Joint(const std::string &name,
+    const PhysicalFrame& parent,
+    const SimTK::Vec3& locationInParent,
+    const SimTK::Vec3& orientationInParent,
+    const PhysicalFrame& child,
+    const SimTK::Vec3& locationInChild,
+    const SimTK::Vec3& orientationInChild,
+    bool reverse) : Super()
+{
+    setNull();
+    constructInfrastructure();
     setName(name);
+    set_reverse(reverse);
+
+    // PARENT TRANSFORM
+    Rotation parentRotation(BodyRotationSequence,
+        orientationInParent[0], XAxis,
+        orientationInParent[1], YAxis,
+        orientationInParent[2], ZAxis);
+    SimTK::Transform parentTransform(parentRotation, locationInParent);
+    PhysicalOffsetFrame parentOffset(parent, parentTransform);
+    parentOffset.setName(parent.getName()+"_offset");
+
+    // CHILD TRANSFORM
+    Rotation childRotation(BodyRotationSequence,
+        orientationInChild[0], XAxis,
+        orientationInChild[1], YAxis,
+        orientationInChild[2], ZAxis);
+    SimTK::Transform childTransform(childRotation, locationInChild);
+    PhysicalOffsetFrame childOffset(child, childTransform);
+    childOffset.setName(child.getName()+"_offset");
+
+    //for debugging color parent green and child blue
+    parentOffset.upd_geometry(0).setColor(SimTK::Vec3(0, 1, 0));
+    parentOffset.upd_geometry(0).set_scale_factors(SimTK::Vec3(0.1, 0.1, 0.1));
+    childOffset.upd_geometry(0).setColor(SimTK::Vec3(0, 0, 1));
+    childOffset.upd_geometry(0).set_scale_factors(SimTK::Vec3(0.1, 0.1, 0.1));
+ 
+
+    // Append the offset frames to the Joints internal list of frames
+    append_frames(parentOffset);
+    append_frames(childOffset);
+
+    updConnector<PhysicalFrame>("parent_frame").set_connectee_name(
+        getName() + "/" + parentOffset.getName() );
+    updConnector<PhysicalFrame>("child_frame").set_connectee_name(
+        getName() + "/" + childOffset.getName() );
```

Comment:
Sorry to pollute this PR with an unrelated comment but I think absolute paths *must* begin with a `/`. Otherwise this ambiguous: do you mean `/jointA/flexion` or `/jointA/jointA/flexion`?